gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageImplTest.java

```diff
@@ -141,8 +142,86 @@ public void writeLargeObject_withSmallUploadChunk() throws IOException {
+
+    int uploadChunkSize = 1024 * 1024;
+    TrackingStorageWrapper<GoogleCloudStorageImpl> trackingGcs =
+        newTrackingGoogleCloudStorage(getOptionsWithUploadChunk(uploadChunkSize));
 
+    int partitionsCount = 1;
+    byte[] partition =
+        writeObject(
+            trackingGcs.delegate,
+            resourceId,
+            /* partitionSize= */ 5 * 1024 * 1024,
+            partitionsCount);
+    List<String> expectedRequests =
+        getExpectedRequestsForCreateObject(resourceId, uploadChunkSize, partitionsCount, partition);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("GET", expectedRequests));
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(1L);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("PUT", expectedRequests));
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(4L);
     assertObjectContent(helperGcs, resourceId, partition, partitionsCount);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("GET", expectedRequests));
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(1L);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("PUT", expectedRequests));
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST_FAILURES)
+                .longValue())
+        .isEqualTo(4L);
+    assertThat(
+            trackingGcs
+                .delegate
+                .getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_POST_REQUEST)
+                .longValue())
+        .isEqualTo(getCountFromExpectedRequests("POST", expectedRequests));
 
     assertThat(trackingGcs.requestsTracker.getAllRequestStrings())
```

Comment:
just to keep the diff consistent, please move the metrics assertions below this

Reply:
Updated